### ISSUE TITLE
ELI5 - polysemy Tactics

### ISSUE THREAD

**Issue by boj** [polysemy-research/polysemy#64]
After #63 @isovector took the time to engage me on a video call and explained how `Tactical` mechanics work. They appear to be something you wouldn't commonly use, but may run into edge cases where they are necessary.

## What (I think) I learned

- When you have some outside `m` context (something in `IO`, a `mtl`-style call, etc) that needs to be ran, you can't incorporate it into your polysemy effects, but you need some way to run it. This is where Tactics come into play
- `pureT`, `runT`, and `bindT` put your external monadic function into the current `Sem` context as some opaque `f` (Sandy said this could be thought of as a monad transformer), which you execute with the `(forall x. Sem r x -> IO x)` runner passed in and composed with your interpreter, like `f' <- bindT f; r .@ runMyInterpreterIO $ f'`
- Sadly you can't treat these things as monads, so no standard combinators available

## Things that may be useful for Sandy to explain

- What problem is being solved with Tactics?
- What is the opaque `f` and why is it synonymous with a monad transformer?
- Why do we have to `fmap` values passed into bound functions? `boundFunc' (param <$ initialstate)`
- What does returning `getInitialStateT` achieve when using an async function?
- Why do we have to call the runner over the same interpreter on an external effect?

**Comment by sevanspowell** [comment]
This is helpful, thank you. +1 those questions.

**Comment by isovector** [comment]
https://reasonablypolymorphic.com/blog/tactics/
